modules/core/src/lib/layer.js

```diff
@@ -513,20 +513,20 @@ export default class Layer extends Component {
 
     // Call subclass lifecycle methods
     this.updateState(updateParams);
+    // End subclass lifecycle methods
 
-    // Render or update previously rendered sublayers
     if (this.isComposite) {
+      // Render or update previously rendered sublayers
       this._renderLayers(updateParams);
-    }
-    // End subclass lifecycle methods
-
-    // Add any subclass attributes
-    this.updateAttributes(this.props);
-    this._updateBaseUniforms();
+    } else {
+      // Add any subclass attributes
+      this.updateAttributes(this.props);
```

Comment:
I wonder if these two lines 
```
this.updateAttributes(this.props);
this._updateBaseUniforms();
```
can be moved inside of `if (this.state.model)` block.